packages/rich-text/src/types.ts

```diff
@@ -26,6 +26,6 @@ export type RichTextValue = {
 	text: string;
 	formats: Array< RichTextFormatList >;
 	replacements: Array< RichTextFormat >;
-	start: number | undefined;
-	end: number | undefined;
+	start: number;
+	end: number;
```

Comment:
Why has this been removed? These are optional.

Reply:
Are they optional properties i.e. a typescript `start?: number` Or optionally set as undefined? Would you mind linking code where this would be set as undefined? My bad I must have totally missed it. Thanks.

Reply:
Probably both. A rich text instance may not have selection

https://github.com/WordPress/gutenberg/blob/f706c86f201f325a4108d74fdf1e0315521db705/packages/block-editor/src/components/rich-text/index.js#L149-L150

https://github.com/WordPress/gutenberg/blob/f706c86f201f325a4108d74fdf1e0315521db705/packages/rich-text/src/component/use-input-and-selection.js#L269

Reply:
O boy, looks like my eyes deceived me. Mind explaining to me the context for which you'd end up with an undefined value? In my usage I have never been able to get an undefined value appear when there is no selection or when there is. Attaching a short video:

https://github.com/WordPress/gutenberg/assets/1588194/2cf0dc4b-5adb-4c98-9dd2-6e3ef52864e2

Would you like me to push another revert commit to put back the optional undefined? Are you good with the other change that was in this commit (the null value for className)?

While I have you attention, through for the wrong reason sadly, has there been any discussion about porting this package to typescript? If so, was there any blockers etc.

Reply:
@ellatrix bumping this. Thanks.

Reply:
All rich text instances have an undefined selection unless it is selected. So if you select one paragraph, all the other paragraph's rich text instances have an undefined selection except for the selected paragraph's.